### ISSUE TITLE
2.8.1 crashes

### ISSUE THREAD

**Issue by specimen151** [arangodb/arangodb#1707]
I upgraded from 2.7.2 (never installed .3) to 2.8.1, and now it crashes. Running on Windows 10 x64 with lots of memory, from command line. Did the --upgrade. Copied the whole var/lib/arangodb directory over, but not the arangodb-apps. The docs just say the "data directory", so I had to guess.

[7656] ERROR Unhandled exception: -1073741819 - will crash now.

I'm using GRAPH_NEIGHBORS, TRAVERSAL w uniqueness. Graph is small (<20 nodes).

I can provide you the code I'm using, if you need.

**Comment by jsteemann** [comment]
Is the crash reproducible with a specific query/operation or does it happen undeterministically? If the former, then if you could provide the exact instructions for reproducing plus any the underlying graph data this would be very helpful. Thanks!

**Comment by specimen151** [comment]
It happens every time. I couldn't see anything in the conversion log, which was empty BTW.
(I looked in ...\AppData\Local\ArangoDB 2.8.1\var\log\arangodb only a .keepme file there). In the web console I have verified that all my collections are there.

Running directly in the web interface I get the following:
```
Query: illegal document handle
stacktrace of offending AQL function: Error: illegal document handle
    at Error (native)
    at Object.AQL_GRAPH_NEIGHBORS (c:\Users\JensPetter\AppData\Local\ArangoDB 2.8.1\share\arangodb\js\server\modules\org\arangodb\aql.js:7864:10)
    at Function.
```
which is maybe a bit more helpful than the previous posted error (from the command line console).
Through the web-interface, Arango Server does not crash. I'm usually going via the REST interface via ArangoDB-NET.

My query is rather large:

```
// Get the whole subtree up til max depth
let subtree = (
    for u in GRAPH_NEIGHBORS('cellGraph', @parentId, 
        {edgeCollectionRestriction:'cellTree', direction:'inbound', maxDepth:@maxDepth})  // find all children
    return u
)

// Find all non-leaf nodes
let nonLeaf = (
for cellId in subtree
    for edge in cellTree
    filter edge._to == cellId
    return distinct cellId
)
// Find leaf nodes by removing non-leaves sub-tree
let leaves = (
for cellId in subtree
filter cellId not in nonLeaf
return cellId
)
// Find all defacto leaves by combining real leaves with cells at exactly max level
let cellsAtLevel2 = (
    for u in GRAPH_NEIGHBORS('cellGraph', @parentId, 
        {edgeCollectionRestriction:'cellTree', direction:'inbound', maxDepth:@maxDepth, minDepth:@maxDepth})
    return u
)
let bottomCells = UNION_DISTINCT(leaves, cellsAtLevel2)
//return {nonLeaf: nonLeaf[*], leaves: leaves[*], cellsAtLevel2:cellsAtLevel2, bottomCells:bottomCells}




// For each cell in bottomCells, we need to run a traversal all the way down
let annotatedBottomCells = (
    for ch in cells
    filter ch._id IN bottomCells
        for c in traversal(cells,cellTree,ch,'inbound',{uniqueness:{vertices:'global'}})
        collect child = ch._id into group
            let num = length(group)
            let works = (
                for w in group 
                return w.c.vertex.Work
            )
            let workDone = (
                for p in group
                return p.c.vertex.Work * p.c.vertex.Progress
            )
            let ParentId = FIRST(
                for edge in cellTree
                filter edge._from == child
                return edge._to
            )
            //let localSubtree = group[*].c.vertex._id
            let localSubtree = (
                for g in group
                filter g.c.vertex._id != g.ch._id
                return {_id:g.c.vertex._id, Work: g.c.vertex.Work, Progress: g.c.vertex.Progress}
            )
            let theChild = FIRST( for ccc in cells filter ccc._id == child return ccc)
            return merge(theChild, {ParentId, group, Subtree:localSubtree, NumDescendants:num, TotalWork:SUM(works), TotalWorkDone: SUM(workDone)})
    )                    
                
                

// Now return these cells + the rest of the subtree of maxDepth
let topCells = (
    for c in cells
    filter c._id in MINUS(subtree,bottomCells)
    let ParentId = FIRST(
        for edge in cellTree
        filter edge._from == c._id
        return edge._to
        )
    return merge(c, {ParentId})
)

return {topCells: topCells, bottomCells: annotatedBottomCells}
```

With bind parameters: 
```
{"parentId":"/cells/ABCRoot",
"maxDepth":"3"}
```

[database-17063398643.zip](https://github.com/arangodb/arangodb/files/120213/database-17063398643.zip)
[logfile-23761893238.zip](https://github.com/arangodb/arangodb/files/120214/logfile-23761893238.zip)

**Comment by specimen151** [comment]
I have tried to create a new database in 2.8.1, fill it with the same data and run the query. Same result. So I guess it is not with the old data.

**Comment by jsteemann** [comment]
Thanks for the data. I can reproduce the crash on Linux as well.

**Comment by jsteemann** [comment]
The crash is fixed in the 2.8 and devel branches now (2.7.x not affected).

The `illegal document handle` error is actually correct from my point of view. It's due to the start id being `"/cells/ABCRoot"`, but it should rather be `"cells/ABCRoot"` (without the forward slash at the start).

**Comment by specimen151** [comment]
That was fast. Thanks.
